src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -100,33 +101,62 @@ def load_config(self):
         else:
             self.metadata_mgr = MetadataManager(self.fs, self.config_path, 0o640)
 
-    def set_attrs(self, path, size, isolate_namespace, pool, uid, gid):
+    def get_attrs(self, pathname):
+        # get subvolume attributes
+        attrs = {}
+        stx = self.fs.statx(pathname,
+                            cephfs.CEPH_STATX_UID | cephfs.CEPH_STATX_GID | cephfs.CEPH_STATX_MODE,
+                            cephfs.AT_SYMLINK_NOFOLLOW)
+
+        attrs["uid"] = int(stx["uid"])
+        attrs["gid"] = int(stx["gid"])
+        attrs["mode"] = int(int(stx["mode"]) & ~stat.S_IFMT(stx["mode"]))
+
+        try:
+            attrs["data_pool"] = self.fs.getxattr(pathname, 'ceph.dir.layout.pool').decode('utf-8')
+        except cephfs.NoData:
+            attrs["data_pool"] = None
```

Comment:
I am not sure how to tackle the mypy errors in these lines, dropping it here if others have any thoughts. Causes failure in the Jenkins make check job.

```
volumes/fs/operations/versions/subvolume_base.py:118: error: Incompatible types in assignment (expression has type "None", target has type "int")
volumes/fs/operations/versions/subvolume_base.py:123: error: Incompatible types in assignment (expression has type "None", target has type "int")
volumes/fs/operations/versions/subvolume_base.py:128: error: Incompatible types in assignment (expression has type "None", target has type "int")
```

Reply:
Initialize attrs["data_pool"] = None before usage. This way the type is set to "Any" and mypy wouldn't complain.

Reference:
1. [mypy-common-issues](https://mypy.readthedocs.io/en/stable/common_issues.html#redefinitions-with-incompatible-types)
2. [Any-type](https://docs.python.org/3/library/typing.html)

Reply:
Maybe go with option 2 in https://stackoverflow.com/questions/43910979/mypy-error-incompatible-types-in-assignment
So something like,
```
attrs = {}  # type: Dict[str, Union[int, str, None]]
```

Reply:
Thanks @kotreshhr @ajarr 

The mypy error is assuming the entire dict stores [str, int], as the first element that I added was an int. Hence the suggestion to initialize it to a None type did not work.

The solution to add a type comment instead worked fine and learnt from @ajarr that this was legal from a [PEP](https://www.python.org/dev/peps/pep-0484/#type-comments) standpoint as well.

Reply:
:+1: